src/library_fs.js

```diff
@@ -118,6 +118,43 @@ FS.staticInit();` +
     filesystems: null,
     syncFSRequests: 0, // we warn if there are multiple in flight at once
 
+    ErrnoError: class extends Error {
+      // We set the `name` property to be able to identify `FS.ErrnoError`
+      // - the `name` is a standard ECMA-262 property of error objects. Kind of good to have it anyway.
+      // - when using PROXYFS, an error can come from an underlying FS
+      // as different FS objects have their own FS.ErrnoError each,
+      // the test `err instanceof FS.ErrnoError` won't detect an error coming from another filesystem, causing bugs.
+      // we'll use the reliable test `err.name == "ErrnoError"` instead
+      constructor(errno) {
+#if ASSERTIONS
+        var message = ERRNO_MESSAGES[errno];
+#else
+        var message = 'FS error';
+#endif
+        super(message);
+        this.name = 'ErrnoError';
```

Comment:
This could be a property on the class perhaps.

Reply:
Sadly we don't support that yet since acorn and closure don't like it.  I was looking into enabling it a few weeks back but ran into issues.

I will add a TODO here.